### PR TITLE
feat: upload file using an s3 client

### PR DESCRIPTION
**Pull request by m90**
Ticket https://phabricator.wikimedia.org/T343112

### CODE REVIEW

**Comment by m90** [comment]
Leaving this a draft until there's a matching chart update.

File: Dockerfile [hunk 1]
```diff
@@ -1,42 +1,35 @@
...
     mydumper=0.9.1-5 \
-    mariadb-client=1:10.1.48-0ubuntu0.18.04.1 \
-  && echo "deb http://packages.cloud.google.com/apt $GCSFUSE_REPO main" \
-    | tee /etc/apt/sources.list.d/gcsfuse.list \
-  && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - \
-  && apt-get update \
-  && apt-get install --yes gcsfuse=0.40.0 --no-install-recommends \
-  && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+    mariadb-client=1:10.1.48-0ubuntu0.18.04.1 && \
+    curl https://dl.min.io/client/mc/release/linux-amd64/mc \
```
Comment (m90) [inline Dockerfile]: This should still be locked down to a specific version.

**Comment by m90** [comment]
Chart update here https://github.com/wbstack/charts/pull/125

File: src/entrypoint.sh [hunk 2]
```diff
@@ -12,9 +12,9 @@ fi
 ## GCS bucket is mounted by chart
 # We can just move the artifacts
 if [ "$DO_UPLOAD" -eq "1" ]; then
-    mv /backups/output/* /mnt/backup-bucket/
+    ./upload.sh
 else
-    echo "Skip uploading..."
+    echo "Skipping upload as DO_UPLOAD is not set..."
```
Comment (m90) [inline src/entrypoint.sh:17]: We could also remove this option now as we can also "upload" in local environments.

File: src/upload.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -e
+mc alias set remote "$STORAGE_ENDPOINT" "$STORAGE_ACCESS_KEY" "$STORAGE_SECRET_KEY" --api "$STORAGE_SIGNATURE_VERSION"
+mc cp --recursive /backups/output remote/"$STORAGE_BUCKET_NAME"
```
Comment (deer-wmde) [inline src/upload.sh]:
```suggestion
mc cp --recursive /backups/output/ remote/"$STORAGE_BUCKET_NAME"
```

This way (with the trailing / for the source directory) it does not create a `output` folder in the bucket which mimics the current behaviour

Comment (m90) [reply src/upload.sh]: Oh, this is a nice find. Thanks. Will also update the test case then.

**Review by deer-wmde — APPROVED** [review]

**Pull request merged**